android/src/main/java/com/reactnativestripesdk/PaymentLauncherFragment.kt

```diff
@@ -5,83 +5,171 @@ import android.view.LayoutInflater
+                   paymentIntentClientSecret: String,
+                   confirmPaymentParams: ConfirmPaymentIntentParams): PaymentLauncherFragment {
+      val paymentLauncherFragment = PaymentLauncherFragment(
+        context,
+        stripe,
+        publishableKey,
+        stripeAccountId,
+        promise,
+        paymentIntentClientSecret = paymentIntentClientSecret,
+        confirmPaymentParams = confirmPaymentParams
+      )
+      addFragment(paymentLauncherFragment, context, promise)
+      return paymentLauncherFragment
+    }
+
+    /**
+     * Helper-constructor used for confirming setup intents
+     */
+    fun forSetup(context: ReactApplicationContext,
+                 stripe: Stripe,
+                 publishableKey: String,
+                 stripeAccountId: String?,
+                 promise: Promise,
+                 setupIntentClientSecret: String,
+                 confirmSetupParams: ConfirmSetupIntentParams): PaymentLauncherFragment {
+      val paymentLauncherFragment = PaymentLauncherFragment(
+        context,
+        stripe,
+        publishableKey,
+        stripeAccountId,
+        promise,
+        setupIntentClientSecret = setupIntentClientSecret,
+        confirmSetupParams = confirmSetupParams
+      )
+      addFragment(paymentLauncherFragment, context, promise)
+      return paymentLauncherFragment
+    }
+
+    /**
+     * Helper-constructor used for handling the next action on a payment intent
+     */
+    fun forNextAction(context: ReactApplicationContext,
+                      stripe: Stripe,
+                      publishableKey: String,
+                      stripeAccountId: String?,
+                      promise: Promise,
+                      handleNextActionClientSecret: String): PaymentLauncherFragment {
+      val paymentLauncherFragment = PaymentLauncherFragment(
+        context,
+        stripe,
+        publishableKey,
+        stripeAccountId,
+        promise,
+        handleNextActionClientSecret = handleNextActionClientSecret,
+      )
+      addFragment(paymentLauncherFragment, context, promise)
+      return paymentLauncherFragment
+    }
+
+    private fun addFragment(fragment: PaymentLauncherFragment, context: ReactApplicationContext, promise: Promise) {
+      (context.currentActivity as? AppCompatActivity)?.let {
+        try {
+          it.supportFragmentManager.beginTransaction()
+            .add(fragment, "payment_launcher_fragment")
+            .commit()
+        } catch (error: IllegalStateException) {
+          promise.resolve(createError(ErrorType.Failed.toString(), error.message))
+        }
+      } ?: run {
+        promise.resolve(createMissingActivityError())
+      }
+    }
+  }
 
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                             savedInstanceState: Bundle?): View {
     paymentLauncher = createPaymentLauncher()
+    if (paymentIntentClientSecret != null && confirmPaymentParams != null) {
```

Comment:
Is it possible to test this? 

I was thinking you could just verify the correct methods are being called with given params. If that's a lot of work feel free to just merge.

Reply:
This is technically covered by tests, just in E2E tests, not unit tests.